### PR TITLE
Update UI to emphasize need to save to remove attachments/criteria

### PR DESCRIPTION
**Pull request by lenglund**
Also includes removing save draft button for non-students (no way to access drafts currently for these folks)

### CODE REVIEW

File: compair/static/modules/assignment/assignment-form-partial.html [hunk 1]
```diff
@@ -45,6 +48,7 @@ <h1 ng-if="method == 'copy'">Duplicate: {{originalAssignment.name}}</h1>
 
         <div>
             <label for="importFile">Related Attachment</label>
+            <p ng-show="!assignment.file && assignment.existingFile" class="alert alert-warning"><strong>Save this assignment to save changes made to your attachment</strong>, or leave this screen to keep the attachment that appeared when you started editing.</p>
```
Comment (kitsook) [inline compair/static/modules/assignment/assignment-form-partial.html]: Ditto. Should the alert also be displayed when:
- adding an attachment to a draft answer that has no attachment?
- rotating an attached image?

File: compair/static/modules/answer/answer-form-partial.html [hunk 2]
```diff
@@ -83,6 +83,7 @@ <h2><i class="fa fa-comments"></i> {{assignment.name}}</h2>
 
         <div>
             <label class="required-star" for="importFile">Answer Attachment and/or Text</label>
+            <p ng-show="!answer.file && answer.existingFile" class="alert alert-warning"><strong>Save<span ng-show="answer.draft"> or submit</span> this answer to save changes made to your attachment</strong>, or cancel this window to keep the attachment that appeared when you started editing.</p>
```
Comment (kitsook) [inline compair/static/modules/answer/answer-form-partial.html]: Should the alert also be displayed when:
- adding an attachment to a draft answer that has no attachment?
- rotating an attached image?

File: compair/static/modules/assignment/assignment-module.js [hunk 3]
```diff
@@ -1402,7 +1402,8 @@ module.controller("AssignmentWriteController",
                 rank_display_limit: null,
                 answer_grade_weight: 1,
                 comparison_grade_weight: 1,
-                self_evaluation_grade_weight: 1
+                self_evaluation_grade_weight: 1,
+                existingFile: false
```
Comment (kitsook) [inline compair/static/modules/assignment/assignment-module.js]: The CI tests are failing.  Need to add this new property (`existingFile: false`) to the test cases too

https://github.com/ubc/compair/blob/05ef114f7038a6e95c069ace46d859e4c9ba7314/compair/static/modules/assignment/assignment-module_spec.js#L1025-L1035

Comment (lenglund) [reply compair/static/modules/assignment/assignment-module.js]: Thank you, can never remember to update the tests...

**Comment by lenglund** [comment]
My logic behind this is: I don't want to tell someone they need to save unless that might be unclear. The need to save is unclear when we visibly remove something from the screen, because a) there is ambiguity as to how deep that removal extends and b) we are relying on the user's memory to recall there used to be something there and they have removed it.

Saving is a more intuitive next step when the user has made a change that is visible, whether rotating the image or updating another input field. These changes still rely on memory of what was, but at least the user can see the change/present state, and it is common for this type of updating to only execute with a save.

Similarly, when the answer is new, my assumption is that the user knows a save has to happen for the answer to be created. Prior to that first save, they should be able to add/remove/edit without the reminders that changes require a save, since anything at that stage requires a save.

File: compair/static/modules/assignment/assignment-form-partial.html (hunk 1 above)
Comment (lenglund) [reply compair/static/modules/assignment/assignment-form-partial.html]: Good questions, address these below.

**Comment by lenglund** [comment]
Add message for rotation

**Comment by lenglund** [comment]
Triggering message during editing for rotating and uploading now. Please review again! Thx.

**Pull request merged**